host/windows/syscall.c

```diff
@@ -324,6 +324,450 @@ __declspec(noreturn) static void _panic(
 **==============================================================================
 */
 
+// Allocates char* string which follows the expected rules for
+// enclaves. Paths in the format
+// <driveletter>:\<item>\<item> -> /<driveletter>/<item>/item>
+// <driveletter>:/<item>/<item> -> /<driveletter>/<item>/item>
+// paths without drive letter are detected and the drive added
+// /<item>/<item> -> /<current driveletter>/<item>/item>
+// relative paths are translated to absolute with drive letter
+// returns null if the string is illegal
+//
+// The string  must be freed
+//
+// we don't handle paths which start with the "\\?\" thing. We never
+// use the 8 bit version of the win32 apis. That said, we get paths in 8 bit
+// characters from the enclave because there is no "wopen" in linux
```

Comment:
The term "8 bit characters" is ambiguous since there are many "8 bit character" encodings, and they are different.  It is not clear whether the implementation below has internationalization bugs.  Normally in Linux "char*" is UTF-8, whereas "char*" in Windows is ANSI code page (which is not interoperable).  So unless you convert the code page to UTF-8, I believe that non-ASCII characters in the path will be passed incorrectly.  I *think* what you mean is that the characters in each label in the path are left in the same format as the caller passed them in, so it's the responsibility of the caller to do any character set conversion.   Assuming that's what you mean, can you update the comment to say that (and not use "8 bit version" or "8 bit characters" as terms).

Reply:
Thanks, comment is updated. According to the implementation, now only ASCII is supported.